src/pages/user/index.tsx

```diff
@@ -25,33 +24,45 @@ const GithubConnectButton: React.FunctionComponent<{
 type ViewerAccount = {
   stripe_customer_id: string
   slug: string
+  subscriptions: any[]
+}
+
+function getAccountWithSubscription(accounts: ViewerAccount[]) {
+  // prefer the account with a subscription, otherwise grab the first, or just
+  // an empty object (which would be an error state, but possible)
+  return (
+    find<ViewerAccount>(
+      accounts,
+      (account: ViewerAccount) => account.subscriptions?.length > 0,
+    ) ||
+    first<ViewerAccount>(accounts) || {slug: ''}
+  )
 }
 
 const User: React.FunctionComponent<
   LoginRequiredParams & {account: ViewerAccount}
 > = () => {
-  const [account, setAccount] = React.useState<any>({})
-  const {stripe_customer_id} = account
+  const [account, setAccount] = React.useState<ViewerAccount>()
   const {viewer, authToken} = useViewer()
   const [progress, setProgress] = React.useState<any>([])
-
   const {email: currentEmail, accounts, providers} = viewer || {}
-  const {slug} = get(accounts, '[0]', {})
+  const {slug} = getAccountWithSubscription(accounts)
   const isConnectedToGithub = providers?.includes('github')
+  const viewerId = viewer?.id
 
   React.useEffect(() => {
-    const loadProgressForUser = async (user_id: number) => {
-      if (user_id) {
-        const {data} = await loadUserProgress(user_id)
+    const loadProgressForUser = async (viewerId: number) => {
+      if (viewerId) {
+        const {data} = await loadUserProgress(viewerId)
         setProgress(data)
       }
     }
 
-    loadProgressForUser(viewer.id)
-  }, [viewer?.id])
+    loadProgressForUser(viewerId)
+  }, [viewerId])
```

Comment:
I find effect dependencies confusing so I like to do this to make them marginally less confusing. YMMV 😬